pkg/matcher/cel.go

```diff
@@ -24,30 +24,66 @@ const (
 	changedFilesTags = "files."
 )
 
+func splitByQuotes(s string) []string {
+	customSplit := func(r rune) bool {
+		return r == '"' || r == '\''
+	}
+	// Split the string based on custom split function
+	return strings.FieldsFunc(s, customSplit)
+}
+
+// getBranchFromExpression function extracts the value associated with the provided branchKey.
+//
+// ex: Given an expression like target_branch == "main", getBranchFromExpression returns "main".
+func getBranchFromExpression(branchKey string) string {
+	parts := strings.Split(branchKey, "==")
+
+	var branch string
+	if len(parts) == 2 {
+		// Trim any leading or trailing white spaces from the value
+		value := strings.TrimSpace(parts[1])
+		sVal := splitByQuotes(value)
+		// Only valid values for example, are target_branch == "main", target_branch == 'main' or source_branch == "*" etc..
+		// Invalid formats include target_branch == ["main"] or target_branch == main.
+		if len(sVal) >= 1 {
```

Comment:
I don't understand the case where the length will be more than 1?

Reply:
This is the scenario

```
    pipelinesascode.tekton.dev/on-cel-expression: |
      ( event == "push" && source_branch == "main" ) && "components/?*/base/**".pathChanged()
```
above `sVal` comes as `[main  )]` and len as 2


```
    pipelinesascode.tekton.dev/on-cel-expression: |
      event == "push" && source_branch == "main" && "components/?*/base/**".pathChanged()
```
above `sVal` comes as `[main]` and len as 1

So checking `len(sVal) >= 1` and sVal[0] will be the branch

Reply:
so i would say better, we remove this prefix and suffix, and keep the remaining string as something to check, there can be case there are in between and end up splitting the value

Reply:
Updated